libecho/src/IAudioConverter.h

```diff
@@ -0,0 +1,36 @@
+#ifndef ECHOCONNECT_I_AUDIO_CONVERTER_H
+#define ECHOCONNECT_I_AUDIO_CONVERTER_H
+
+/**
+ * @brief Interface for classes which converts between audio and data.
+ */
+
+#include <vector>
+
+template <typename T>
+class IAudioConverter {
+public:
+    IAudioConverter(QAudioFormat inputFormat, QAudioFormat outputFormat, double windowLength)
+        : inputFormat(inputFormat), outputFormat(outputFormat), windowLength(windowLength){};
+
+    std::vector<char> encode(std::vector<uint8_t> data) {
+        std::vector<T> result;
+        for (auto byte : data) {
+            auto encoded_byte = encode_byte(byte);
+            result.insert(result.end(), encoded_byte.begin(), encoded_byte.end());
+        }
+        return std::vector<char>(result.data(), result.data() + result.size());
```

Comment:
niezbyt mi się podoba pomysł, żeby wrzucać wszystko do vectora, a następnie tworzyć na jego podstawie nowy wyłącznie po to, aby przekonwertować typy

Reply:
To jest tymczasowe rozwiązanie, będę pracować nad sensownym podejściem do konwersji typów w tym miejscu. Plusem takiego rozwiązania jest to, że nie mamy reinterpret_castów.

Reply:
reinterpret_cast na char* akurat ma poprawnie zdefiniowane w standardzie zachowanie, więc nie widzę powodu, żeby go na siłę unikać (szczególnie, że w tym przypadku program prawdopodobnie mógłby działać szybciej)

Reply:
Rozumiem, wezmę to pod uwagę i rozwiążę w przyszłym PR kiedy to będę się zajmować ładną konwersją typów. Nie będę tego zmieniać na ten moment, ponieważ będzie to wymagało niepotrzebnej, tymczasowej zmiany interfejsu.